### ISSUE TITLE
wirePopulateStringTags() does not work with subfields

### ISSUE THREAD

**Issue by noelboss** [processwire/processwire-issues#940]
I can't seem to make subfields in wirePopulateStringTags() work. 

### Optional: Steps that explain the enhancement

Tried everything, nothing worked. For example, consider the following:

```
dump(
	wirePopulateStringTags(
		'{page.title}', 
		new WireData([
			'page' => wire('pages')->get(1)
		]), 
		[
			'recursive' => true,
			'removeNullTags' => false
		]
	)
);

dump(
	wirePopulateStringTags(
		'{page.title}',
		[
			'page' => wire('pages')->get(1),
		],
		[
			'recursive'      => true,
			'removeNullTags' => false,
		]
	)
);
```

### Output:
![Screenshot 2019-07-25 at 13 39 35](https://user-images.githubusercontent.com/114080/61871656-a6c03600-aee1-11e9-82ae-03dbacb46ad2.png)

The explanations from the doc page are not conclusive or helpful beyond the simple user:
https://processwire.com/api/ref/functions/wire-populate-string-tags/

Is it me? Or it might just simply not work (looking at the source) – but it would be amazing if it did – would save us from some simple plugins…

**Comment by Toutouwai** [comment]
The title isn't a subfield of a page. And I think `wire('page')->get(1)` might be a typo.
It's working for me like this:
![2019-07-26_004351](https://user-images.githubusercontent.com/1538852/61875313-75298800-af3e-11e9-8b8b-821050006a62.png)

**Comment by teppokoivula** [comment]
Looks like this might be related to a request I submitted a while ago: https://github.com/processwire/processwire-requests/issues/309.

It's true that you can access values like file.description or images.count if you're passing a Page object to wirePopulateStringTags(), but if you want to pass a WireData that contains multiple keys (such as in the code examples above), you currently can't access these values.

In my example I've created a wrapper object for WireData to add this behaviour (a modified version of WireData::get()).

**Comment by noelboss** [comment]
@Toutouwai page<strike>s</strike> was a typo, indeed... Fixed it in the initial post…

Consider this example (more like what I want then the above example)…

```
dump(
	wirePopulateStringTags(
		'Home: {home.title} Admin: {admin.title}',
		new WireData([
			'home'  => wire('pages')->get('id=1'),
			'admin' => wire('pages')->get('id=2'),
		]),
		[
			'recursive'      => true,
			'removeNullTags' => false,
		]
	)
);

dump(
	wirePopulateStringTags(
		'Home: {home.title} Admin: {admin.title}',
		[
			'home'  => wire('pages')->get('id=1'),
			'admin' => wire('pages')->get('id=2'),
		],
		[
			'recursive'      => true,
			'removeNullTags' => false,
		]
	)
);
```

**Comment by Toutouwai** [comment]
I agree this would be useful, but it's a request not a bug, because the docs say that "$vars may also be an object, in which case values will be pulled as properties of the object". So it's object, singular, and the subfields feature applies only to this singular object: "The tag parser can also handle subfields and OR tags, if $vars is an object that supports that."

**Comment by noelboss** [comment]
Hmm... I find this very it confusing… Especially with the examples: it states for example;  "For instance {products.title}" – Thats exactly what I would expect from my snippet above. Also the docs mention {products.title} but below is a foo bar example. Not helpful in my opinion. So, how can {products.title} work but I'm not able to construct this object in a way, that it also works with WireData or with a page?

**Comment by ryancramerdesign** [comment]
Fixed. When `$vars` is a `WireData`, dot notation now uses `getDot()` instead of `get()` (which doesn't traverse dots). When `$vars` is a plain array, we split on the first dot, pull the sub-object from the array, and delegate to `getDot()` or `WireData::_getDot()` depending on type.

Note: when constructing a `WireData` to pass as `$vars`, use the `WireData()` helper function rather than `new WireData()` — the constructor doesn't accept an array argument, but the helper does.

-Claude
